### PR TITLE
chore(combobox): refactor to use useCombobox

### LINKED ISSUE #14684: Refactor ComboBox to use `useComboBox` instead of `<Downshift>` component
Downshift plans to move forward with their React Hooks API and deprecate and remove `<Downshift>` from the API:

* https://github.com/downshift-js/downshift/issues/1233

We need to refactor our ComboBox to no longer use `<Downshift>` and instead use the `useComboBox` hook. As part of this we'll need to ensure downstream consumers aren't impacted at all and nothing of significance changes. If we can't contain this without a breaking change, we'll need to put the refactor behind a feature flag (using the same component name) and slate it for v12.

### LINKED ISSUE #15927: [Bug]: Combo box incorrectly resets to previously selected value (or no-value) when a click occurs INSIDE the text input
### Package

@carbon/react

### Browser

Chrome

### Package version

v1.52.0

### React version

v18.2.0

### Description

When using the Combo Box, it is expected that if you type in text and click away, it should reset to a valid state which would be Empty, or the previously selected value.

However, this behavior should not occur when clicking INSIDE the search input itself. For example, lets say I am typing into the Combo Box, and realize I had a misspelling. I click to move the cursor, and the text resets to the last valid state. This feels like a bug and is not a great user experience. 

This can be reproduced in the story book: https://react.carbondesignsystem.com/?path=/docs/components-combobox--overview

### Reproduction/example

https://react.carbondesignsystem.com/?path=/docs/components-combobox--overview

### Steps to reproduce

1. Go to: https://react.carbondesignsystem.com/?path=/story/components-combobox--default
2. Type "Optyon" into the box
3. Realize you spelled it wrong and place the cursor before the "y"
4. Notice the value is cleared out ): 

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Application/PAL

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

### PR DESCRIPTION
**Pull request by preetibansalui**
Closes #14684 
Closes https://github.com/carbon-design-system/carbon/issues/15927

Refactored the code to use usecombobox

#### Changelog

**Changed**

Code refactored

#### Testing / Reviewing

It should work in similar ways as existing combobox.

### CODE REVIEW

File: packages/react/src/components/ComboBox/ComboBox.tsx [hunk 1]
```diff
@@ -855,8 +837,9 @@ ComboBox.propTypes = {
   /**
    * Additional props passed to Downshift
    */
-  // @ts-ignore
-  downshiftProps: PropTypes.shape(Downshift.propTypes),
+  downshiftProps: PropTypes.object as React.Validator<
+    UseComboboxProps<unknown>
+  >,
```
Comment (tay1orjones) [inline packages/react/src/components/ComboBox/ComboBox.tsx:845]: I'm concerned this could be a breaking change

Comment (preetibansalui) [reply packages/react/src/components/ComboBox/ComboBox.tsx:845]: Any suggestions please? 
Functionality wise it's working fine although with 'onStateChange' like what we were using this prop earlier.

Comment (tay1orjones) [reply packages/react/src/components/ComboBox/ComboBox.tsx:845]: Yeah you're right this looks fine, my mistake

**Comment by tay1orjones** [comment]
@tw15egan @guidari This one's ready for review

It looks like this solves https://github.com/carbon-design-system/carbon/issues/15927 too. 

I've closed the original PR, https://github.com/carbon-design-system/carbon/pull/15904

**Review by guidari — CHANGES_REQUESTED** [review]
Hey Preeti! Sorry for the late review on this.

A few things I noticed
1 - The ComboBox only opens when the icon is clicked. it should open if is clicked in the anywhere inside the input.
2 - When opening the Combobox with keyboard and searching for "3" (which is the disabled option) and after that pressing Enter, the Combobox is closed. It should stay open. 
Also looks like is not announcing the disabled option in the VO when searching.

I'm happy to pair up to check any of those cases, if you want.

File: packages/react/src/components/ComboBox/ComboBox.tsx [hunk 2]
```diff
@@ -421,65 +405,71 @@ const ComboBox = forwardRef(
...
-          }
-          break;
-        case clickButton:
-        case clickItem:
-        case unknown:
-          setHighlightedIndex(getHighlightedIndex(changes));
-          break;
-      }
-    };
+            break;
```
Comment (guidari) [inline packages/react/src/components/ComboBox/ComboBox.tsx]: We can remove that.
```suggestion
```

**Comment by tay1orjones** [comment]
> A few things I noticed
> 1 - The ComboBox only opens when the icon is clicked. it should open if is clicked in the anywhere inside the input.
> 2 - When opening the Combobox with keyboard and searching for "3" (which is the disabled option) and after that pressing Enter, the Combobox is closed. It should stay open.

It would be awesome if as these are fixed, new test cases could be written to cover this functionality and ensure it continues to work as expected

**Review by guidari — APPROVED** [review]
LGTM! 🚀
Let's open an issue to track the VO not reading the disabled item

**Review by tay1orjones — APPROVED** [review]
LGTM once the conflicts are fixed 🎉

**Pull request merged**